library/src/main/java/com/novoda/downloadmanager/LiteDownloadService.java

```diff
@@ -46,10 +48,14 @@ public IBinder onBind(Intent intent) {
 
     @Override
     public void download(final DownloadBatch downloadBatch, final DownloadBatchCallback callback) {
-        callback.onUpdate(downloadBatch.status());
```

Comment:
are you sure about this? if we get here and the status is downloaded, the callback won't be notified...

Reply:
Interesting, does this only work because of the `getAllBatchStatuses`... I'll take another look 💯